src/helper/service.cpp

```diff
@@ -72,6 +78,38 @@ service::ptr service::from_settings(const service_identifier &id,
         std::chrono::milliseconds{rc_settings.poll_interval});
 }
 
+void service::handle_error()
+{
+    rc_action_ = [this] { discover(); };
+    interval_ = std::chrono::duration_cast<std::chrono::milliseconds>(
+        poll_interval_ * pow(2, std::min(errors_, max_increment)));
```

Comment:
The only concern is that if we change the poll interval, the maths won't work.

Reply:
I see what you mean. I'll create a ticket for this.